src/ListenerService.ts

```diff
@@ -14,54 +14,39 @@
  * limitations under the License.
  */
 
-import * as assert from 'assert';
 import * as Promise from 'bluebird';
-import {EventEmitter} from 'events';
-import {ConnectionHeartbeatListener} from './core/ConnectionHeartbeatListener';
-import {Member} from './core/Member';
 import HazelcastClient from './HazelcastClient';
 import {HazelcastError} from './HazelcastError';
-import {ClientConnection} from './invocation/ClientConnection';
+import {ClientConnection} from './network/ClientConnection';
 import {ClientEventRegistration} from './invocation/ClientEventRegistration';
 import {Invocation} from './invocation/InvocationService';
 import {RegistrationKey} from './invocation/RegistrationKey';
 import {ListenerMessageCodec} from './ListenerMessageCodec';
-import {copyObjectShallow, DeferredPromise} from './Util';
+import {DeferredPromise} from './Util';
 import {UuidUtil} from './util/UuidUtil';
 import {ILogger} from './logging/ILogger';
 
-export class ListenerService implements ConnectionHeartbeatListener {
+export class ListenerService {
     private client: HazelcastClient;
-    private internalEventEmitter: EventEmitter;
     private logger: ILogger;
     private isShutdown: boolean;
```

Comment:
nit: this field is never read after these changes and, probably, can be removed along with the `shutdown()` method.